src/compiler/checker.ts

```diff
@@ -26461,14 +26451,9 @@ namespace ts {
                 checkGrammarForGenerator(node);
             }
 
-            const type = getTypeOfSymbol(getMergedSymbol(node.symbol));
-            if (isTypeAny(type)) {
-                return type;
-            }
-
             contextuallyCheckFunctionExpressionOrObjectLiteralMethod(node, checkMode);
 
-            return type;
+            return getTypeOfSymbol(getSymbolOfNode(node));

```

Comment:
👍  on using `getSymbolOfNode`, but why'd we eschew the `isTypeAny` bail?

Reply:
Because there's nothing gained from that extra code. The following code paths already ignore functions whose type turn out to be any.